ldap/servers/slapd/conntable.c

```diff
@@ -125,12 +125,17 @@ connection_table_new(int table_size)
 {
     Connection_Table *ct;
     int i = 0;
+    int ct_list = 0;
+    int free_idx = 0;
     ber_len_t maxbersize = config_get_maxbersize();
 
     ct = (Connection_Table *)slapi_ch_calloc(1, sizeof(Connection_Table));
     ct->size = table_size;
-    ct->c = (Connection *)slapi_ch_calloc(1, table_size * sizeof(Connection));
-    ct->fd = (struct POLL_STRUCT *)slapi_ch_calloc(1, table_size * sizeof(struct POLL_STRUCT));
+    ct->list_num = 2;
+    ct->list_size = table_size/ct->list_num;
```

Comment:
If the user has configured a list_size that is not a multiple of table_size or list_num, we'll end up with orhapned connections that can't be accessed. So I think there needs to be a way to assert that these are valid sizes.

Reply:
Good point.

Reply:
Nice catch!     Having:
 ct->list_size = table_size/ct->list_num + 1; /* +1 to avoid rounding issue */
should simply solve that issue 
No orphan connection because ct->list_size * ct->list_num > table_size 
And we do not really care about having list size slightly too long.